azure/notebooks/azure_mnmg_xgboost.ipynb

```diff
@@ -0,0 +1,1287 @@
+    "\n",
+    "For the purposes of this demo, we will use a part of the NYC Taxi Dataset (only the files of 2014 calendar year will be used here). The goal is to predict the fare amount for a given trip given the times and coordinates of the taxi trip. We will download the data from [Azure Open Datasets](https://azure.microsoft.com/en-us/services/open-datasets/), where the dataset is publicly hosted by Microsoft.\n",
+    "\n",
+    "---\n",
+    "\n",
+    "**NOTE:**  In this notebook, we will explore two possible ways to use `dask-cloudprovider` to run our workloads on dask on Azure VM clusters:\n",
+    "\n",
+    "1. [Option 1](#-Option-1:-Use-an-Azure-marketplace-VM.-): Using an Azure Marketplace VM image made available for free from NVIDIA. The RAPIDS container will be subsequently downloaded once the VMs start up.\n",
+    "2. [Option 2](#-Option-2:-Set-up-an-Azure-Customized-VM.-): Using `packer` to create a custom VM image to be used in the cluster. The image will include the RAPIDS container. Having the container already inside the image should speed up the process of provisioning the cluster.\n",
+    "\n",
+    "#### Either use Option 1 or use Option 2. Not both.\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## <span style=\"color:#8735fb; font-size:22pt\"> Step 0: Import necessary packages. </span>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Uncomment the following and install some libraries at the beginning. \n",
+    "# # If azureml is not present install azureml-core. \n",
+    "# # Further opendatasets is in preview mode hence it is not available in the main sdk and needs to be installed separately.\n",
+    "# # Installing azureml-opendatasets with for some weird reason installs an old version of Pandas 1.0.0, which we will upgrade to 1.2.4 to work with RAPIDS. \n",
+    "# # We do the same for the numpy and scipy libraries that are also downgraded by azureml-opendatasets.\n",
+    "# ! pip install \"dask-cloudprovider[azure]\"\n",
+    "# ! pip install \"dask-cloudprovider[azure]\" --upgrade\n",
+    "# ! pip install azureml-core\n",
+    "# ! pip install azureml-opendatasets\n",
+    "# ! pip install azureml-telemetry\n",
+    "# ! pip install pandas==1.2.4 # reverting pandas to 1.2.4\n",
+    "# ! pip install numpy==1.20.2 # reverting numpy to 1.20.2\n",
+    "# ! pip install scipy==1.6.0 # reverting scipy to 1.16.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Client, wait, get_worker\n",
+    "from dask_cloudprovider.azure import AzureVMCluster\n",
+    "import dask_cudf\n",
+    "from azureml.opendatasets import NycTlcYellow\n",
+    "from dask_ml.model_selection import train_test_split\n",
```

Comment:
`ModuleNotFoundError: No module named 'dask_ml'`

Need to install dask-ml

Reply:
Or can we use cuml instead?

Reply:
But `dask-ml` is already preinstalled in rapidsai/rapidsai:cuda11.2-runtime-ubuntu18.04-py3.8 . It is a bit strange that you got an error. Also, I tested with `cuml`, but it ran into errors while splitting into test and train. I presume `cuml` does not work here since the data is already distributed,. Hence I am using the dask version.

Reply:
I just pulled rapidsai and rapidsai-core and neither of them seem to have dask-ml, these are the dask packages: `dask dask-core dask-cuda dask-cudf`. This is also based on user's local env so need to mention that in the beginning as well, similar to the RF notebook (which also looks like doesn't install dask-ml? @Nanthini10 )

Reply:
Alrighty looks like it's available in runtime but not core images. Can we just link to the README that mentions local env/setup? Same for the RF notebook (since we now have many examples with different setup in the README :) )

Reply:
Updated the README for both notebooks.